apps/publisher/themes/default/js/taxonomy-browser.js

```diff
@@ -0,0 +1,326 @@
+
+        $(this).closest('li').siblings().siblings().each(function () {
+            $(this).removeClass('active');
+            $('button', this).remove();
+        });
+        $(this).closest('li:not(.back)').addClass('active');
+
+        $('[data-window]:gt(' + dataWindow + ')').each(function () {
+            $(this).find('li').removeClass('active');
+            windowObject[$(this).data('parent')] = $(this);
+            $(this).remove();
+        });
+
+        if (windowObject[element]) {
+            $(TAXONOMY_SELECT).append(windowObject[element].first());
+            delete windowObject[element];
+            updateTaxonomyWindow(dataWindow);
+        } else {
+            loadTaxonomyChildren(root, element, dataWindow);
+        }
+
+        // Updating taxonomy breadcrumb
+        $(BREADCRUMB_SELECTOR).html($(BREADCRUMB_SELECTOR + ' > li').first());
+        $(COLUMN_SELECTOR + ' ul > li').each(function () {
+            if ($(this).hasClass('active')) {
+                $(BREADCRUMB_SELECTOR).append('<li>' + $('a', this).html() + '</li>');
+            }
+        });
+    });
+
+    $(TAXONOMY_SELECT).on('click', COLUMN_SELECTOR + ' li.back > a', function (e) {
+        e.preventDefault();
+
+        var dataWindow = $(this).closest(COLUMN_SELECTOR).data('window'),
+            windowsToHide = ( (dataWindow - 1) + columnsCount );
+
+        // Hide windows that are overflowing
+        $('[data-window]:gt(' + (windowsToHide - 1) + ')').hide();
+        $('[data-window=' + dataWindow + '] > ul > li.back').hide();
+        $('[data-window=' + ( dataWindow - 1 ) + ']').fadeIn();
+    });
+
+    $(TAXONOMY_SELECT).on('click', COLUMN_SELECTOR + ' li.leaf > a', function (e) {
+        e.preventDefault();
+        displayValue.length = 0;
+        var elemParent = $(this).closest('li');
+        elemParent.siblings().each(function () {
+            $(this).removeClass('active');
+            $('button', this).remove();
+        });
+        $(this).closest('li:not(.back)').addClass('active');
+
+        // Updating taxonomy breadcrumb
+        $(BREADCRUMB_SELECTOR).html($(BREADCRUMB_SELECTOR + ' > li').first());
+        $(COLUMN_SELECTOR + ' ul > li').each(function () {
+            if ($(this).hasClass('active')) {
+                displayValue.push($('a', this).text());
+                $(BREADCRUMB_SELECTOR).append('<li>' + $('a', this).html() + '</li>');
+            }
+        });
+
+        //Appending the add button to leaf nodes
+        if (selectedTaxonomy.indexOf($(this).attr('href')) < 0) {
+            elemParent.append(' <button class="btn btn-add"><span>Add</span>'
+                + '<span class="icon fw fw-stack"><i class="fw fw-add fw-stack-1x"></i>'
+                + '<i class="fw fw-circle-outline fw-stack-2x"></i></span></button>');
+        }
+    });
+
+    $(TAXONOMY_SELECT).on('click', COLUMN_SELECTOR + ' li.leaf > button', function (e) {
```

Comment:
@splinter I have already added a comment here in 47a37a0dd9e9a41be0dcced7f68650998e776226. Also I have to mention that this file structure has been changed in later PRs